### ISSUE TITLE
[Question] Change root

### ISSUE THREAD

**Issue by ghost** [oooutlk/trees#9]
Is it possible to change the root of the tree to another node? The rest of the Tree above should be deleted.  For example
..................
.     0         .
.   /   \        .
.  1     4     .
.            \   .
.              7.
.................
Change to root to Number 4
.............
.     4     .
.       \   .
.        7  .
.............

**Comment by oooutlk** [comment]
The [`Node::detach()`](https://docs.rs/trees/0.4.1/trees/node/struct.Node.html#method.detach) method can make a freestanding `Tree` "4( 7 )" out of "0( 1 4( 7 ) )".

```rust
let mut tree = tr(0) /tr(1) /( tr(4)/tr(7) );
let mut new_tree = tree.back_mut().unwrap().detach();
std::mem::swap( &mut tree, &mut new_tree );
assert_eq!( tree, tr(4)/tr(7) );
```
